qiskit/result/result.py

```diff
@@ -38,34 +49,112 @@ class Result:
             each experiment success)
         results (list[ExperimentResult]): corresponding results for array of
             experiments of the input qobj
+        date (str): date of the experiment
+        header(dict): experiment header
+        kwargs: generic keyword arguments. These will be stored in the metadata field.
     """
 
     _metadata = {}
 
     def __init__(
         self,
-        backend_name,
-        backend_version,
-        qobj_id,
-        job_id,
-        success,
-        results,
+        *args,
```

Comment:
This might be confusing for people because the signature went from saying `backend_name`, `backend_version`, `qobj_id`, `job_id`, `success`, and `results` were all required arguments to no longer specifying that anywhere. The condition hasn't changed, but nothing is indicating that anymore.

Reply:
I think it's also important to note that 2.0 will get back the named arguments, this is really just necessary to be able to raise the deprecation warning.

Reply:
Yeah I would add it to the docstring, I think that's all we can do in this situation